spec/unit/mixin/softlayer_metadata_spec.rb

```diff
@@ -0,0 +1,71 @@
+#
+# Author:: Alexey Karpik <alexey.karpik@rightscale.com>
+# Author:: Peter Schroeter <peter.schroeter@rightscale.com>
+# Author:: Stas Turlo <stanislav.turlo@rightscale.com>
+# Copyright:: Copyright (c) 2010-2014 RightScale Inc
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require File.expand_path(File.dirname(__FILE__) + '/../../spec_helper.rb')
+require 'ohai/mixin/softlayer_metadata'
+
+
+describe ::Ohai::Mixin::SoftlayerMetadata do
+
+  let(:mixin) {
+    mixin = Object.new.extend(::Ohai::Mixin::SoftlayerMetadata)
+    mixin
+  }
+
+  def make_request(item)
+    "/rest/v3.1/SoftLayer_Resource_Metadata/#{item}"
+  end
+
+  def make_res(body)
+    double("response", {:body => body, :code => "200"})
+  end
+
+
+  context 'fetch_metadata' do
+    it "creates empty softlayer hash on api query errors" do
```

Comment:
As I have explained below, even if we could not get data from metadata server, we logged error, we still need to have softlayer node, so other plugins get detect on which cloud they are running on.